### PR TITLE
CORDA-876 MockNetwork no longer leaks serialization env if init fails

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 1]
```diff
@@ -221,11 +222,16 @@ class MockNetwork(private val cordappPackages: List<String>,
...
-        val notaryInfos = generateNotaryIdentities()
-        // The network parameters must be serialised before starting any of the nodes
-        networkParameters = NetworkParametersCopier(testNetworkParameters(notaryInfos))
-        notaryNodes = createNotaries()
+        try {
+            filesystem.getPath("/nodes").createDirectory()
+            val notaryInfos = generateNotaryIdentities()
+            // The network parameters must be serialised before starting any of the nodes
+            networkParameters = NetworkParametersCopier(testNetworkParameters(notaryInfos))
+            notaryNodes = this.createNotaries()
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: "this" should be redundant, but there's an unused warning if i leave it out

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 2]
```diff
@@ -131,7 +131,8 @@ class MockNetwork(private val cordappPackages: List<String>,
                   servicePeerAllocationStrategy: InMemoryMessagingNetwork.ServicePeerAllocationStrategy = defaultParameters.servicePeerAllocationStrategy,
                   private val defaultFactory: (MockNodeArgs) -> MockNode = defaultParameters.defaultFactory,
                   initialiseSerialization: Boolean = defaultParameters.initialiseSerialization,
-                  private val notarySpecs: List<NotarySpec> = defaultParameters.notarySpecs) {
+                  private val notarySpecs: List<NotarySpec> = defaultParameters.notarySpecs,
+                  createNotaries: MockNetwork.() -> List<StartedNode<MockNode>> = MockNetwork.Companion::createNotariesImpl) {
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: We're leaking this parameter into the public API. Is there no other way to cause the init block to throw an exception? Perhaps have an internal static boolean field `shouldThrow` (I know, I know, really horrible). I'll try to think of a better suggestion.

Comment (andr3ej) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: oops, forgot about the api. the way i used to solve this sort of problem in java is to simply override the createNotaries method in the test. that would mean making MockNetwork open, but i'm not sure that's a big deal. what do you think?

Comment (shamsasari) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: I think that works.

Comment (andr3ej) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: Done

**Comment by andr3ej** [comment]
i think i broke tests by moving that line in stopNodes, will fix in a bit

**Pull request merged**